app/src/lib/api.ts

```diff
@@ -229,10 +238,15 @@ export class API {
   /** Fetch a commit from the repository. */
   public async fetchCommit(owner: string, name: string, sha: string): Promise<IAPICommit | null> {
     try {
-      const response = await this.request('GET', `repos/${owner}/${name}/commits/${sha}`)
+      const path = `repos/${owner}/${name}/commits/${sha}`
+      const response = await this.request('GET', path)
+      if (response.status === HttpStatusCode.NotFound) {
+        log.warn(`fetchCommit: '${path}' returned a 404`)
+        return null
+      }
       return parsedResponse<IAPICommit>(response)
     } catch (e) {
-      log.warn(`fetchCommit: not found '${owner}/${name}@${sha}'`, e)
+      log.warn(`fetchCommit: returned an error '${owner}/${name}@${sha}'`, e)
```

Comment:
@iAmWillShepherd yeah, I'm mentioning `error` here but I don't think it's worth cluttering up our log files with these. `warn` will just output to the dev tools, which I think is fine for now.